src/components/tooltip/tooltip.js

```diff
@@ -3,392 +3,405 @@
+      // panelPosition API to build the panel's position.
+      panelPosition = $mdPanel.newPanelPosition()
+          .relativeTo(parent)
+          .addPanelPosition(
+            $mdPanel.xPosition[position.x],
+            $mdPanel.yPosition[position.y]
+          );
+
+      // If the panel has already been created, add the new origin class to
+      // the panel element and update it's position with the panelPosition.
+      if (panelRef && panelRef.panelEl) {
+        panelRef.panelEl.addClass(origin);
+        panelRef.updatePosition(panelPosition);
+      }
+    }
+
+    function bindEvents() {
+      // Add a mutationObserver where there is support for it and the need
+      // for it in the form of viable host(parent[0]).
       if (parent[0] && 'MutationObserver' in $window) {
-        // use an mutationObserver to tackle #2602
+        // Use a mutationObserver to tackle #2602.
         var attributeObserver = new MutationObserver(function(mutations) {
-          if (mutations.some(function (mutation) {
-              return (mutation.attributeName === 'disabled' && parent[0].disabled);
-            })) {
-              $mdUtil.nextTick(function() {
-                setVisible(false);
-              });
+          if (isDisabledMutation()) {
+            $mdUtil.nextTick(function() {
+              setVisible(false);
+            });
           }
         });
 
-        attributeObserver.observe(parent[0], { attributes: true});
+        attributeObserver.observe(parent[0], {
+          attributes: true
+        });
       }
 
-      // Store whether the element was focused when the window loses focus.
-      var windowBlurHandler = function() {
-        elementFocusedOnWindowBlur = document.activeElement === parent[0];
-      };
+      elementFocusedOnWindowBlur = false;
 
-      var elementFocusedOnWindowBlur = false;
+      $$mdTooltipRegistry.register('scroll', windowScrollEventHandler, true);
+      $$mdTooltipRegistry.register('blur', windowBlurEventHandler);
+      $$mdTooltipRegistry.register('resize', debouncedOnResize);
 
-      function windowScrollHandler() {
-        setVisible(false);
-      }
+      scope.$on('$destroy', onDestroy);
 
-      $$mdTooltipRegistry.register('scroll', windowScrollHandler, true);
-      $$mdTooltipRegistry.register('blur', windowBlurHandler);
-      $$mdTooltipRegistry.register('resize', debouncedOnResize);
+      // To avoid 'synthetic clicks', we listen to mousedown instead of
+      // 'click'.
+      parent.on('mousedown', mousedownEventHandler);
+      parent.on(ENTER_EVENTS, enterEventHandler);
 
-      scope.$on('$destroy', function() {
-          $$mdTooltipRegistry.deregister('scroll', windowScrollHandler, true);
-          $$mdTooltipRegistry.deregister('blur', windowBlurHandler);
-          $$mdTooltipRegistry.deregister('resize', debouncedOnResize);
+      function isDisabledMutation() {
+        mutations.some(function(mutation) {
```

Comment:
@bradrich This throws an error. Seems like you forgot to pass in the `mutations`.